java-checks/src/test/java/org/sonar/java/checks/spring/SpringBeansShouldBeAccessibleCheckTest.java

```diff
@@ -68,4 +76,19 @@ void testSpringBootApplication() {
       .verifyNoIssues();
   }
 
+  @Test
+  void testSpringBootApplicationWithAnnotation() {
+    final String testFolder = BASE_PATH + "springBootApplication/";
+    List<String> files = Arrays.asList(
+      testSourcesPath(testFolder + "thirdApp/SpringBootApp.java"),
+      testSourcesPath(testFolder + "thirdApp/domain/SomeClass.java"),
+      testSourcesPath(testFolder + "thirdApp/controller/Controller.java"));
+
+    JavaCheckVerifier.newVerifier()
+      .onFiles(files)
+      .withCheck(new SpringBeansShouldBeAccessibleCheck())
+      .verifyIssues();
```

Comment:
What bothers me when testing multiples files is that if you put the `Noncompliant` comment in `SomeClass.java`, line 6 and remove it from `Controller.java`, the test will pass despite the fact that the issue is not raised where you would expect.
I would rather test one file at a time, and set the classPath (`withClassPath`) if needed.

Reply:
(See related ticket describing the problem: [SONARJAVA-3643](https://jira.sonarsource.com/browse/SONARJAVA-3643))

Reply:
Well, its indeed a limitation that I had to fight already when moving the other test files. However, your solution won't work either. The rule is based on an `EndOfAnalysisCheck` which requires analyzing all the files... So using the classpath won't work. The collects first annotations and at the end of the analysis, report.

Reply:
Okay, thanks for the prescision, I missed that indeed.